javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -100,21 +99,63 @@ public Node(final int beginLine, final int beginColumn, final int endLine, final
 	public abstract <A> void accept(VoidVisitor<A> v, A arg);
 
 	/**
+	 * Returns the begin position of this node.
+	 *
+	 * @return the begin position, or <code>null</code> if the position has not been set
+	 */
+	public Position getBegin() {
+		return begin;
+	}
+
+	/**
+	 * Sets the begin position of this node.
+	 *
+	 * @param begin
+	 *              the end position
+	 */
+	public void setBegin(Position begin) {
+		this.begin = begin;
+	}
+
+	/**
+	 * Returns the end position of this node.
+	 *
+	 * @return the end position, or <code>null</code> if the position has not been set
+	 */
+	public Position getEnd() {
+		return end;
+	}
+
+	/**
+	 * Sets the end position of this node.
+	 *
+	 * @param end
+	 *            the end position
+	 */
+	public void setEnd(Position end) {
+		this.end = end;
+	}
+
+	/**
 	 * Return the begin column of this node.
 	 * 
 	 * @return the begin column of this node
+	 * @deprecated Use getBegin().column
 	 */
+	@Deprecated()
 	public final int getBeginColumn() {
-		return beginColumn;
+		return begin == null ? 0 : begin.column;
```

Comment:
Actually it returns 0. But maybe that would be a good thing to return -1 in the new implementation. I wonder how many people rely on this to differentiate parsed node from manually created ones... (actually nobody I guess!)
What do you think ?